lib/host.rb

```diff
@@ -511,31 +511,24 @@ def delete(file, opts={})
         if bad_files.include? file
           raise "should not remove file named '#{file}'"
         end
-
-        r = "-r"
+        meth = :rm
       else
-        r = ""
+        meth = :remove
       end
       file = shell_escape file
       if opts[:home] && ! file.start_with?('/','\\')
         # relative to host local user home directory
         # relies strongly on bad_files checking above
         file = '~/' + file
       end
-      if opts[:raw]
-        exec_method = :exec_raw
-      elsif opts[:admin]
-        exec_method = :exec_admin
+      if File.file? file
+        FileUtils.public_send(method, file, force: true)
+        check_cmd = File.file? file
       else
-        exec_method = :exec
+        FileUtils.public_send(:remove_dir, file, force: true)
```

Comment:
need to check `File.dir?` here otherwise if the path doesn't exist at all the `force: true` will eat the error and we won't know that the path is neither file nor dir because it doesn't exist.

Reply:
@rbbratta the point of `delete` is to remove the target, if the target doesn't exists already, then we don't really care if there's error or not... the ultimate goal is `target` resource is not there.

Reply:
we need to follow current convention. Honestly I didn't check what it is. But we just keep existing behavior.

Reply:
the system library has it all  `:remove_entry` or `:remove_entry_secure` will remove anything. :smiley: 

https://rubydoc.info/stdlib/fileutils/FileUtils#remove_entry_secure-class_method